packages/plugins/rpc-manager/src/lib/index.ts

```diff
@@ -20,25 +22,78 @@ export default class RpcManager {
   private events: EmbarkEvents;
   public _accounts: any[] | null = null;
   public _nodeAccounts: any[] | null = null;
+  private blockchainConfig: EmbarkConfig["blockchainConfig"];
   constructor(private readonly embark: Embark) {
     this.events = embark.events;
     this.logger = embark.logger;
     this.rpcModifierEvents = new Events() as EmbarkEvents;
+    this.blockchainConfig = this.embark.config.blockchainConfig;
     this.init();
   }
 
-  private async init() {
+  protected get web3() {
+    return (async () => {
+      if (!this._web3) {
+        let provider;
+        let testsCliOptions;
+        let isTest = false;
 
-    this.rpcModifierEvents.setCommandHandler("nodeAccounts:updated", this.updateAccounts.bind(this));
-    this.rpcModifierEvents.setCommandHandler("nodeAccounts:added", async (addedNodeAccount: any, cb: Callback<null>) => {
+        await this.events.request2("blockchain:started");
+
+        if (this.embark.currentContext.includes(constants.contexts.test)) {
+          isTest = true;
+          testsCliOptions = await this.events.request2("tests:cli:options");
+        }
+        if (isTest && testsCliOptions.node === constants.blockchain.vm) {
+          provider = await this.events.request2("blockchain:client:vmProvider");
+        } else {
+          let endpoint = this.blockchainConfig.endpoint;
+          if (isTest && testsCliOptions.node) {
+            endpoint = testsCliOptions.node;
```

Comment:
Sadly, this could cause issues, because tests enable users to modify the endpoint inside each file.
eg:
If `--node` is `"http://localhost:1234"`, then the first test will use that endpoint to connect, but then, the second test could have a blockchain config with `endpoint: "http://localhost:9876`, then it would connect to that new one.

As you can see, with you changes, it would always connect to `1234`, because it's the `--node`, but it can be overwritten.

Anyway, the tests actually change the blockchainConfigs, so what you will find in `blockchainConfig` should always be up to date, so just reseting `_web3` one `blockchain:started` should work and then, just check what's in the configs.

Reply:
Good point -- I'll wait til the ganache PR is merged, then remove this whole chunk of code and replace it with:
```
protected get web3() {
    return (async () => {
      if (!this._web3) {
        await this.events.request2("blockchain:started");
        const provider = await this.events.request2("blockchain:node:provider", "ethereum");
        this._web3 = new Web3(provider);
      }
      return this._web3;
    })();
  }
```
I'll also revert the introduced test cli option stuff.

Much simpler!

Reply:
Yeah totally 👍

Reply:
Updated.